torchmultimodal/models/flava/flava_model.py

```diff
@@ -636,24 +631,32 @@ def forward(self, x: torch.Tensor) -> torch.Tensor:
 class DalleVAEEncoder(nn.Module, PretrainedMixin):
     def __init__(
         self, image_size: Union[int, Tuple[int, int]] = 112, pretrained: bool = True
-    ):
+    ) -> None:
         super().__init__()
         self.image_size = image_size
         self.encoder = DalleEncoder()
         if pretrained:
             self.load_model()
 
-    def load_model(self):
+    def load_model(
+        self,
+        pretrained_url: str = None,
+        load_state_dict: bool = False,
+        state_dict_key: Optional[str] = None,
+    ) -> Dict[str, Any]:
         # TODO (T116682215): Network error due to FLAVA model relying on access to openAI
         encoder = super().load_model(
-            "https://cdn.openai.com/dall-e/encoder.pkl", load_state_dict=False
+            "https://cdn.openai.com/dall-e/encoder.pkl", load_state_dict=load_state_dict
         )
-        self.encoder.load_state_dict(encoder.state_dict())
+        if isinstance(encoder, dict):
+            self.encoder.load_state_dict(encoder)
+        else:
+            self.encoder.load_state_dict(encoder.state_dict())
         return self.state_dict()
 
     def get_codebook_indices(self, images: Tensor) -> Tensor:
         z_logits = self.encoder(images)
-        return torch.argmax(z_logits, axis=1)
+        return torch.argmax(z_logits, dim=1)
 
     def get_codebook_probs(self, images: Tensor) -> Tensor:
```

Comment:
why are we touching this logic

Reply:
mypy gave an error on this, and the [documentation](https://pytorch.org/docs/stable/generated/torch.argmax.html) also uses `dim` instead of `axis`